src/main/java/se/liu/ida/hefquin/engine/wrappers/lpgwrapper/impl/LPG2RDFConfigurationReader.java

```diff
@@ -0,0 +1,112 @@
+import java.net.URISyntaxException;
+
+public class LPG2RDFConfigurationReader {
+
+    public static LPG2RDFConfigurationReader instance = new LPG2RDFConfigurationReader();
+
+    protected LPG2RDFConfigurationReader(){}
+
+    public static LPG2RDFConfigurationImpl readFromFile( final String filename ) {
+        return instance.parseLPG2RDFConf(filename);
+    }
+
+    public static LPG2RDFConfigurationImpl readFromModel(final Model lpg2Rdf ) {
+        return instance.parseLPG2RDFConf(lpg2Rdf);
+    }
+
+    public LPG2RDFConfigurationImpl parseLPG2RDFConf( final String filename ) {
+        final Model lpg2Rdf = RDFDataMgr.loadModel(filename);
+        return parseLPG2RDFConf(lpg2Rdf);
+    }
+
+    public LPG2RDFConfigurationImpl parseLPG2RDFConf(final Model lpg2Rdf ) {
+        final NodeMapping nodeMapping;
+        final String label;
+
+        final ResIterator lpg2rdfConfigs = lpg2Rdf.listResourcesWithProperty(RDF.type, LPG2RDF.LPGtoRDFConfiguration);
+
+        if (!lpg2rdfConfigs.hasNext()){
+            throw new IllegalArgumentException("LPGtoRDFConfiguration is required!");
+        }
+        else {
+            final Resource lpg2rdfConfig = lpg2rdfConfigs.next();
+
+            verifyLPG2RDFConfiguration(lpg2rdfConfig);
+
+            if(!lpg2rdfConfig.hasProperty(LPG2RDF.labelPredicate)){
+                throw new IllegalArgumentException("labelPredicate is required!");
+            }
+            final RDFNode labelPredicateObj = lpg2rdfConfig.getProperty(LPG2RDF.labelPredicate).getObject();
+            String url;
+            try {
+                url = labelPredicateObj.asResource().getURI();
+            }
+            catch (ResourceRequiredException re){
+                throw new IllegalArgumentException("labelPredicate is invalid!");
+            }
+            label = verifyExpectedURI(url).toString();
+
+            final Resource nodeMappingResource = lpg2rdfConfig.getProperty(LPG2RDF.nodeMapping).getResource();
+            final RDFNode nodeMappingResourceType = lpg2Rdf.getRequiredProperty(nodeMappingResource, RDF.type).getObject();
+
+            if ( nodeMappingResourceType.equals(LPG2RDF.IRIBasedNodeMapping) ) {
+                if(!nodeMappingResource.hasProperty(LPG2RDF.prefixOfIRIs)){
+                    throw new IllegalArgumentException("prefixOfIRIs is required!");
+                }
+                final RDFNode prefixOfIRIObj = nodeMappingResource.getProperty(LPG2RDF.prefixOfIRIs).getObject();
+                final String prefixOfIRIUrl = verifyExpectedURI(prefixOfIRIObj.asResource().getURI()).toString();
+                nodeMapping = new NodeMappingToURIsImpl(prefixOfIRIUrl);
+            }
+            else if ( nodeMappingResourceType.equals(LPG2RDF.BNodeBasedNodeMapping) ) {
+                nodeMapping = new NodeMappingToBNodesImpl();
+            }
+            else {
+                throw new IllegalArgumentException("Type of Resource in NodeMapping is not defined.");
```

Comment:
To make it easier for users, increase the information content of this error message by including--in parenthesis within the message--the URI of `nodeMappingResourceType`).

Reply:
This comment is not clear to me, and I am not able to get your point, can you provide more information about this comment?

Reply:
The if-else block before considers two types of things: instances of the RDF class `IRIBasedNodeMapping` and instances of the RDF class `BNodeBasedNodeMapping`. This exception here is for things that are not in any of these two classes (at least not according to an explicit `rdf:type` triple in the given RDF description). So, in the given RDF description they may have `rdf:type` triple(s) with some other class URIs, which was probably a mistake by the user when he/she wrote the RDF description. Now, to make it easier for the user to understand where and what the mistake in the RDF description is, the error message should also mention the (unexpected/unsupported) class URI of the corresponding `rdf:type` triple in the given RDF description.